src/lib/CriteriaResolution.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.13;
+
+struct CriteriaResolver {
```

Comment:
What is the purpose of a "Criteria"?  Maybe add a brief explanation to this `struct` and to the `_validateCriteria` function below?

Reply:
Noted, it's only vaguely explained in other files.